### ISSUE TITLE
Partial namespace in SchemaId

### ISSUE THREAD

**Issue by henkosch** [domaindrivendev/Swashbuckle#333]
I have many complex models in my API.
They have full type names like this:

```
MyProject.Api.Interface.Models.Category1.Type1
MyProject.Api.Interface.Models.Category1.Type2
MyProject.Api.Interface.Models.Category2.Type1
MyProject.Api.Interface.Models.Category2.Type2
```

As you can see I have two different types with the class name Type1, so currently I'm using UseFullTypeNameInSchemaIds() configuration to avoid collision.

But I want to make my SchemaIds shorter like this:

```
Category1.Type1
Category1.Type2
Category2.Type1
Category2.Type2
``` 

How could I achieve this?
I have tried to create a custom ISchemaFilter, but I could not find a way to change the SchemaId.

**Comment by domaindrivendev** [comment]
Unfortunately the only way to do this right now would be with a DocumentFilter. You would have to replace the "definitions" dictionary, transforming the keys accordingly as well as iterating and updating all Schema refs throughout the document.

You may be quicker to submit a PR. In this, you could update the generator options to include a strategy (lambda) for schemaIds then wire up a default implementation that uses the existing approach

**Comment by SquigNob** [comment]
I also need this feture so have added PR  #377

**Comment by gambrose** [comment]
I would like this feature but would like the strategy to take the Type and return a string for the Id.

PR #377 only lets you change the transformed id.

My use case is to combine class names for nested types but ignore namespaces. But you might want to use attributes etc to define your ids.

**Comment by gambrose** [comment]
@domaindrivendev do you have a date when 5.2.2 is likely to be released?
I've been holding off using this feature until it lands in the NuGet feed.

**Comment by domaindrivendev** [comment]
Available in v5.2.2. See SchemaId option in the config API
